From 8596e40564ee9e883d144906cdc5dc8272445e93 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:45:29 -0400
Subject: [x86] pci: Remove acpi_get_pxm() usage

Message-id: <20140806034529.1438.1336.stgit@gir.stowe>
Patchwork-id: 87422
O-Subject: [RHEL7.1 PATCH 13/69] x86/PCI: Remove acpi_get_pxm() usage
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: ab6ffce35b2cd1f558fce954d38a7ee7ffa9c415

commit ab6ffce35b2cd1f558fce954d38a7ee7ffa9c415
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Fri Jan 24 14:40:46 2014 -0700

    x86/PCI: Remove acpi_get_pxm() usage

    The PCI host bridge code doesn't care about _PXM values directly; it only
    needs to know what NUMA node the hardware is on.

    This uses acpi_get_node() directly and removes the _PXM stuff.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/pci/acpi.c b/arch/x86/pci/acpi.c
index 755eae7..53ba77a 100644
--- a/arch/x86/pci/acpi.c
+++ b/arch/x86/pci/acpi.c
@@ -480,9 +480,6 @@ struct pci_bus *pci_acpi_scan_root(struct acpi_pci_root *root)
  struct pci_bus *bus = NULL;
  struct pci_sysdata *sd;
  int node;
-#ifdef CONFIG_ACPI_NUMA
- int pxm;
-#endif
 
  if (pci_ignore_seg)
   domain = 0;
@@ -494,12 +491,7 @@ struct pci_bus *pci_acpi_scan_root(struct acpi_pci_root *root)
   return NULL;
  }
 
- node = NUMA_NO_NODE;
-#ifdef CONFIG_ACPI_NUMA
- pxm = acpi_get_pxm(device->handle);
- if (pxm >= 0)
-  node = pxm_to_node(pxm);
-#endif
+ node = acpi_get_node(device->handle);
  if (node == NUMA_NO_NODE)
   node = x86_pci_root_bus_node(busnum);
 
@@ -570,15 +562,8 @@ struct pci_bus *pci_acpi_scan_root(struct acpi_pci_root *root)
    pcie_bus_configure_settings(child);
  }
 
- if (bus && node != NUMA_NO_NODE) {
-#ifdef CONFIG_ACPI_NUMA
-  if (pxm >= 0)
-   dev_printk(KERN_DEBUG, &bus->dev,
-       "on NUMA node %d (pxm %d)\n", node, pxm);
-#else
+ if (bus && node != NUMA_NO_NODE)
   dev_printk(KERN_DEBUG, &bus->dev, "on NUMA node %d\n", node);
-#endif
- }
 
  return bus;
 }
-- 
1.7.1